Sources/SwiftDiscord/DiscordEndpoint.swift

```diff
@@ -17,6 +17,49 @@
 
 import Foundation
 
+/**
+ * An HTTP content-type.  Common options are available as enum values, but if you need something special, use .other("My-Special-Type")
+ */
+public enum HTTPContentType: CustomStringConvertible {
+    case json
+    case other(String)
+
+    public var description: String {
+        switch self {
+        case .json:
+            return "application/json"
+        case let .other(type):
+            return type
+        }
+    }
+}
+
+/**
+ * An HTTP Request Method.  This includes any associated data.
+ */
+public enum HTTPMethod {
```

Comment:
Maybe move this inside of DiscordEndpoint and rename it EndpointRequest? Maybe it would make sense to move `createRequest` to be a method of this and pass in the endpoint?